packages/wrangler/src/deployment-bundle/source-maps.ts

```diff
@@ -0,0 +1,136 @@
+import fs from "node:fs";
+import path from "node:path";
+import type { BundleResult, SourceMapMetadata } from "./bundle";
+import type { CfModule, CfWorkerSourceMap } from "./worker";
+
+/**
+ * Loads source maps that appear in the given build output.
+ */
+export function loadSourceMaps(
+	main: CfModule,
+	modules: CfModule[],
+	bundle: Partial<BundleResult>
+): CfWorkerSourceMap[] {
+	const { sourceMapPath, sourceMapMetadata } = bundle;
+	if (sourceMapPath && sourceMapMetadata) {
+		// This worker was bundled by Wrangler, so we already know where
+		// the source map is located.
+		return loadSourceMap(main, sourceMapPath, sourceMapMetadata);
+	} else {
+		// Don't know where source maps are located, so need to find
+		// them by scanning the contents of the user-specified modules.
+		return scanSourceMaps([main, ...modules]);
+	}
+}
+
+interface SourceMapV3 {
```

Comment:
Elsewhere in Wrangler we actually use a library for this:

https://github.com/cloudflare/workers-sdk/blob/bd00dc827769145df8d42a677393173bd751e643/packages/wrangler/src/dev/inspect.ts#L14

Reply:
Updated to use the type from this library